components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -305,6 +323,7 @@ IJsonSerializer jsonSerializer
                         var newJobId = newJob.Job.Id;
                         if (jobIdDict.ContainsKey(newJobId)) {
                             assignedJobs[jobIdDict[newJobId].Item2] = newJob;
+                            publisherDiagnosticInfo.Add(newJobId, _publisherDiagnosticInfo[newJobId]);
```

Comment:
Since entries are added to `_publisherDiagnosticInfo` dictionary only after a heartbeat is reported we might have a situation where for an assigned job we have not yet received a heartbeat and accompanying diagnostics information. So we have to first check that an entry with `newJobId` key exists in `_publisherDiagnosticInfo` before accessing it.